Cenium.Reservations/Cenium.Reservations.Client.Windows/Actions/CancelAction.cs

```diff
@@ -0,0 +1,237 @@
+﻿/*
+ * Copyright (c) Cenium AS. All Right Reserved
+ *
+ * This source is subject to the Cenium License.
+ * Please see the License.txt file for more information.
+ * All other rights reserved.
+ * 
+ * http://www.cenium.com
+ * 
+ * Change History:
+ * 
+ * User        Date          Comment
+ * ----------- ------------- --------------------------------------------------------------------------------------------
+ * Malikshi.P  12/31/2021    Created
+ */
+
+
+using Cenium.Framework.Client;
+using Cenium.Framework.Client.AppResources.UI;
+using Cenium.Framework.Client.Model;
+using Cenium.Framework.Client.Windows;
+using Cenium.Framework.Client.Windows.Pages;
+using Cenium.Framework.Client.Windows.Pages.Actions;
+using Cenium.Framework.ComponentModel;
+using System;
+using System.ComponentModel;
+using System.Windows;
+
+namespace Cenium.Reservations.Client.Windows.Actions
+{
+    /// <summary>
+    /// Explain the purpose of the class here
+    /// </summary>
+    /// 
+    [RegisterActionType("reservations.cancelaction")]
+    public class CancelAction : AbstractActionHandler
+    {
+        private Record _record = null;
+
+        /// <summary>
+        /// Initializes a new instance of the InitializeAction class
+        /// </summary>
+        public CancelAction()
+        {
+            IsActionEnabled = true;
+            IsActionItemRequired = true;
+        }
+
+        /// <summary>
+        /// Evaluates the conditions when the ActionItem property is set.
+        /// </summary>
+        /// <returns>true if the conditions are fullfilled; otherwise false</returns>
+        protected override bool EvaluateConditions()
+        {
+            bool test = base.EvaluateConditions();
+            if (test)
+            {
+                if (ClientConnector.Current.CheckUserAccess("Reservations/Reservation/Cancel"))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Action to execute when generating prices.
+        /// </summary>
+        protected override void OnExecute()
+        {
+            var rec = GetRecord();
+            if (rec == null)
+            {
+                MessageBox.Show("Unable to find record.");
+                return;
+            }
+            else if (rec.State == RecordState.Added || rec.IsDirty)
+            {
+                MessageBox.Show("Please save changes.");
```

Comment:
use WIndowManager implementations for displaying errors/warnings/confirmations etc. IFx this through the client code

ex:
WindowManager.ShowInfomationMessageDialog()
WindowManager.SHowErrorMesssageDialog()
WindowManager.ShowConfirmationMessageDialog()

Use them with LanguageConstants, instead of hardcoding message.